### PR TITLE
Docs update migration guide 784

### LINKED ISSUE #784: Create a migration guide to explain how to move from V1 to V2
**Is your feature request related to a problem? Please describe.**

With the release of the SDK V2, a migration guide for migrating scripts developed against SDK V1 would be useful.

**Describe the solution you'd like**

The most appropriate location for this material would likely be a subsection of https://planet-sdk-for-python-v2.readthedocs.io/en/latest/upgrading/

**Describe alternatives you've considered**

Description of differences inline in tutorials and other content.

**Additional context**

Requested by PM.

### PR DESCRIPTION
**Pull request by JuLeeAtPlanet**
**Related Issue(s):**

Closes #784

**Proposed Changes:**

For inclusion in changelog (if applicable):

n/a

Not intended for changelog:

1. Documentation now includes upgrading from v1 to v1

**Diff of User Interface**

n/a

**PR Checklist:**

- [x] This PR is as small and focused as possible
- [] If this PR includes proposed changes for inclusion in the changelog, the title of this PR summarizes those changes and is ready for inclusion in the Changelog.
- [] I have updated docstrings for function changes and docs in the 'docs' folder for user interface / behavior changes
- [x] This PR does not break any examples or I have updated them

**(Optional) @mentions for Notifications:**
@cholmes

### CODE REVIEW

File: docs/get-started/upgrading.md [hunk 1]
```diff
@@ -1,20 +1,85 @@
...
+# Process result
+```
+
+For more information about Session, refer to the [Python SDK User Guide](../../python/sdk-guide/#session).
+
+## Asynchronous Methods
+
+With the V1 client, all communication was synchronous. Asynchronous bulk support was provided with the `downloader` module. There was no built-in support for polling when an order was ready to download or tracking when an order was downloaded.
+
+In V2, all `*Client` methods (for example, `DataClient().quick_search`, `OrderClient().create_order`) are asynchronous. Any functions that call such methods must include `async` in their definition. To invoke asynchronous methods from synchronous code, you can wrap the async method calls in `asyncio.run()`. The following is an example of using async with session.
```
Comment (jreiberkyle) [inline docs/get-started/upgrading.md:38]:
```suggestion
In V2, all `*Client` methods (for example, `DataClient().search_aiter`, `OrderClient().create_order`) are asynchronous. Any functions that call such methods must include `async` in their definition. To invoke asynchronous methods from synchronous code, you can wrap the async method calls in `asyncio.run()`. The following is an example of using async with session.
```

Comment (jreiberkyle) [reply docs/get-started/upgrading.md:38]: see #785 for function name update

File: docs/get-started/upgrading.md [hunk 2]
```diff
@@ -1,20 +1,85 @@
...
+```console
+planet.api.ClientV1().quick_search(filters.build_search_request(all_filters, ["PSScene"]))
+```
+
+Is now
+
+```console
+async with Session() as session: 
+planet.DataClient(session).quick_search(["PSScene"], all_filters)
+```
```
Comment (jreiberkyle) [inline docs/get-started/upgrading.md:79]:
```suggestion
```python
async with Session() as session: 
    items_aiter = planet.DataClient(session).search_aiter(["PSScene"], all_filters)
```
```

File: docs/get-started/upgrading.md [hunk 3]
```diff
@@ -1,20 +1,85 @@
...
+The Data API portion of SDK V2 is quite similar to V1, although some filters have been renamed for consistency (also reference the note on imports):
+
+* `date_range` to `date_range_filter`
+* `geom_filter` to `geometry_filter`
+* `and_filter` and `or_filter` now takes a list of filters instead of multiple arguments, so just wrap the contents in `[]` 
+* `permissions_filter` is now split into `permissions_filter` and `asset_filter`, reflecting a recent change in API structure. If you were using this previously, you’ll want to convert the old `permissions_filter` into an `asset_filter` (this also involves changing the filter values, e.g. `assets.ortho_analytic_8b_sr:download` will become `ortho_analytic_8b_sr`) and adding an empty `permissions_filter`.
+
+`filters.build_seach_request` no longer exists, and has instead been integrated into the replacement for `client.quick_seach`. For example:
+
+```console
```
Comment (jreiberkyle) [inline docs/get-started/upgrading.md:70]:
```suggestion
```python
```

**Review by jreiberkyle — APPROVED** [review]
looks amazing Julee! Suggested a few changes to reflect the current api

**Review by cholmes — APPROVED** [review]
Looks good! There are a few things I'd consider adding, but all of them could come later, so marking this as approved so we can get this in. Happy to help ticket my suggestions if they sound good and we'd do them later.

* Advantages of upgrading - would be good to somewhere articulate what you 'gain' from upgrading, why this is a win. Particularly the bulk requests you can do with async, and the great traffic shaping to maximally utilize our API's. I think @jreiberkyle is planning a blog post on this, so we could just link out to that from here. Or could try to explain in a line or two.
* Additional functionality - this is related to the above, but potentially slightly different. One big one is support for Subscriptions API. We probably should also mention that basemaps and analytics aren't yet supported, so we don't have an 'upgrade path' for those and recommend sticking with v1. But I'm also wondering if we have like nicer helper functions? And like orders says 'orders api capabilities in v1 were quite primitive' - we could explain more of what v2 can help with, that are beyond 'primitive'.
* Links to more async info. Perhaps we even work towards a dedicated section somewhere on async, since we know it's the trickiest aspect. Where we link to some relevant tutorials on async in general, and also link to our examples. We could at least link to the few examples / notebooks we have that show a few different async examples.

Again, this more than satisfies the core need to point people on v1 at a clear upgrade guide, and these are all 'nice to have', and may well better belong in other documents.

**Pull request merged**